node/src/store/sled_store.rs

```diff
@@ -0,0 +1,565 @@
+use std::convert::Infallible;
+use std::ops::Deref;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use celestia_types::hash::Hash;
+use celestia_types::ExtendedHeader;
+use directories::ProjectDirs;
+use sled::transaction::{ConflictableTransactionError, TransactionError};
+use sled::{Db, Error as SledError, Transactional, Tree};
+use tempdir::TempDir;
+use tendermint_proto::Protobuf;
+use tokio::task::spawn_blocking;
+use tokio::task::JoinError;
+use tracing::debug;
+
+use crate::store::Store;
+use crate::store::{Result, StoreError};
+
+const HEAD_HEIGHT_KEY: &[u8] = b"KEY.HEAD_HEIGHT";
+const HASH_TREE_ID: &[u8] = b"HASH";
+const HEIGHT_TO_HASH_TREE_ID: &[u8] = b"HEIGHT";
+
+#[derive(Debug)]
+pub struct SledStore {
+    inner: Arc<Inner>,
+}
+
+#[derive(Debug)]
+struct Inner {
+    db: Db,
+    headers: Tree,
+    height_to_hash: Tree,
+}
+
+impl SledStore {
+    pub async fn new(network_id: String) -> Result<Self> {
+        spawn_blocking(move || {
+            let Some(project_dirs) = ProjectDirs::from("co", "eiger", "celestia") else {
+                return Err(StoreError::BackingStoreError(
+                    "Unable to get system cache path to open header store".to_string(),
+                ));
+            };
+            let mut db_path = project_dirs.cache_dir().to_owned();
+            db_path.push(network_id);
+
+            let db = sled::open(db_path)?;
+            Self::init(db)
+        })
+        .await?
+    }
+
+    pub async fn new_temp() -> Result<Self> {
+        spawn_blocking(move || {
+            let tmp_path = TempDir::new("celestia")?.into_path();
+
+            let db = sled::Config::default()
+                .path(tmp_path)
+                .temporary(true)
+                .create_new(true) // make sure we fail if db is already there
+                .open()?;
+            Self::init(db)
+        })
+        .await?
+    }
+
+    pub async fn new_in_path<P>(path: P) -> Result<Self>
+    where
+        P: Into<PathBuf> + Send + 'static,
```

Comment:
The convention is `AsRef<Path>`. After that you can convert it to PathBuf within the method. Remove `Send + 'static` also.

Reply:
👍 reverted back to AsRef<Path>